include/util.h

```diff
@@ -0,0 +1,37 @@
+#ifndef __BM_UTIL_H__
+#define __BM_UTIL_H__
+
+#include <stdbool.h>
+
+typedef enum {
+    BmOK = 0,
+    BmEPERM = 1,
+    BmENOENT = 2,
+    BmEIO = 5,
+    BmENXIO = 6,
+    BmEAGAIN = 11,
+    BmENOMEM = 12,
+    BmEACCES = 13,
+    BmENODEV = 19,
+    BmEINVAL = 22,
+    BmENODATA = 61,
+    BmENONET = 64,
+    BmEBADMSG = 74,
+    BmEMSGSIZE = 90,
+    BmENETDOWN = 100,
+    BmETIMEDOUT = 110,
+    BmECONNREFUSED = 111,
+    BmEHOSTDOWN = 112,
+    BmEHOSTUNREACH = 113,
+    BmEALREADY = 114,
+    BmEINPROGRESS = 115,
+    BmECANCELED = 125,
+} BMErr;
```

Comment:
I decided to use standard POSIX errors as a starting point for the bm errors that could exist. When we add more functionality this can be scaled to inform integrators of larger issues.